app/src/main/java/com/example/xpvehicles/activities/VehicleDetailsActivity.java

```diff
@@ -203,12 +294,16 @@ private void calculateOrderTotal(int numberOfRentDays) {
     private Boolean checkValidDates() {
         Boolean validDates = true;
         if (pickupDate == null) {
-            detailsPickupDateOTF.setError("Enter a pickup date");
-            validDates = false;
+            detailsPickupDateOTF.setError("Enter a valid date");
+            validDates =  false;
         }
         if (returnDate == null) {
-            detailsReturnDateOTF.setError("Enter a return date");
-            validDates = false;
+            detailsReturnDateOTF.setError("Enter a valid date");
+            validDates =  false;
+        }
+        if (pickupDate != null && returnDate != null && returnDate.getTime() - pickupDate.getTime() < 0) {
```

Comment:
Do we allow same date return?

Reply:
Yes, that is where int conversionDifference = 1 ; comes into play.